rails_programming/rails_basics/importmap.md

```diff
@@ -33,30 +38,30 @@ Import maps are essentially string substitution for "bare module specifiers". If
+When creating a new Rails 7+ application `importmap-rails` will be automatically added to your Gemfile. `importmap-rails` is doing a few things for you. First the `<%= javascript_importmap_tags %>` line is added to `views/layouts/application.html.erb` which will set up the JSON configuration inside of a `<script type="importmap">` tag. Then the [es-module-shim](https://github.com/guybedford/es-module-shims) is loaded. Finally, the application entry point is imported via `<script type="module">import "application"</script>`. Inside your `app/javascript/application.js` you set up the application by importing any modules that have been defined in the import map. It is good to use logical names that match the package names used by NPM, that way if you want to start transpiling or bundling your code you won't have to change any module imports.
 
-You start the setup via the configuration file located `config/importmap.rb`. This file is automatically reloaded in development when anything changes. That being said, if you remove pins and need them gone from the rendered importmap or list of preloads you MUST restart the server. 
+You start the setup via the configuration file located `config/importmap.rb`. This file is automatically reloaded in development when anything changes. That being said, if you remove pins and need them gone from the rendered importmap or list of preloads you MUST restart the server.
 
 ### Import Maps with NPM packages
 
 `importmap-rails` is designed to be used with JavaScript CDNs for NPM package dependencies.
 
-You can use the `./bin/importmap` command that's added as part of the install to `pin`, `unpin`, or `update` npm packages in your import map. This command resolves the package dependencies and adds the pins to your `config/importmap.rb`. Here is an example:
+You can use the `./bin/importmap` command that's added as part of the install to `pin`, `unpin`, or `update` npm packages in your import map. This command resolves the package dependencies and adds the pins to your `config/importmap.rb`.
 
-```JS
+~~~JS
 ./bin/importmap pin react react-dom
 Pinning "react" to https://ga.jspm.io/npm:react@17.0.2/index.js
```

Comment:
I know it's not part of this PR, but I think using a package the student can follow and code along with would be better for the code examples. Perhaps something like lodash? they can pin it and use one the lodash string functions to console.log something. I'm a little anxious of these examples going over students heads if they can't see it working for themselves.

But that could be a follow up if you think it's worth doing? it doesn't have to stop this from going out.

Reply:
@KevinMulhern That's a good point mate. Usually I wouldn't mind reworking it but I think this could be a wider thing we look into as part of the rails_path ownership where we ensure all examples are runnable in some way for those that want to try.